revisions.py

```diff
@@ -0,0 +1,482 @@
+# -*- coding: utf-8 -*-
+"""Functions for revision management."""
+
+__copyright__ = 'Copyright (c) 2019-2020, Utrecht University'
+__license__   = 'GPLv3, see LICENSE'
+
+import os
+import time
+
+import folder
+import meta_form
+
+from util import *
+from util.query import Query
+
+__all__ = ['api_revisions_restore',
+           'api_revisions_search_on_filename',
+           'api_revisions_list',
+           'rule_revisions_clean_up']
+
+
+@rule.make(inputs=range(2), outputs=range(2, 3))
+def rule_revisions_clean_up(ctx, bucketcase, endOfCalendarDay):
+    """
+    Step through entire revision store and apply the chosen bucket strategy
+    :param bucketcase       multiple ways of cleaning up revisions can be chosen.
+    :parem endOfCalendarDay if zero, system will determine end of current day in seconds since epoch (1970-01-01 00:00 UTC)
+    """
+    zone = user.zone(ctx)
+    revision_store = '/' + zone + constants.UUREVISIONCOLLECTION
+
+    if user.user_type(ctx) == 'rodsadmin':
+        msi.set_acl(ctx, "recursive", "admin:own", user.full_name(ctx), revision_store)
+        msi.set_acl(ctx, "recursive", "inherit", user.full_name(ctx), revision_store)
+
+    end_of_calender_day = int(endOfCalendarDay)
+    if end_of_calender_day == 0:
+        end_of_calendar_day = calculate_end_of_calendar_day(ctx)
+
+    # get definition of buckets
+    buckets = revision_bucket_list(ctx, bucketcase)
+    log.write(ctx, buckets)
```

Comment:
There is still some debug logging left. To prevent cluttering the logs we can remove these or change them to log.debug if they are valuable. Log.debug statements will only show on development instances.